cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/JpaRestApiClient.java

```diff
@@ -0,0 +1,294 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparest;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Fail.fail;
+import static uk.gov.homeoffice.digital.sas.jparest.controller.enums.RequestParameter.TENANT_ID;
+
+import io.restassured.response.Response;
+import io.restassured.specification.QueryableRequestSpecification;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.SpecificationQuerier;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.http.HttpHeaders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * The purpose of this class is to understand the protocol implemented by the JpaRest library. At
+ * this moment in time it knows how to perform CRUD operations on the endpoints exposed but when
+ * authorisation is implemented in the API this class will be updated to add authentication to the
+ * requests being made.
+ */
+@Component
+public class JpaRestApiClient {
+
+  public static final String API_ROOT_PATH = "/resources/";
+  public static final String TENANT_ID_SYSTEM_PROPERTY_NAME = "cucumber.jparest.tenantId";
+  private static final String TENANT_ID_SYSTEM_PROPERTY_EL
+      = "#{systemProperties['" + TENANT_ID_SYSTEM_PROPERTY_NAME + "']}";
+
+  /**
+   * Injecting tenantId parameter as a system property to avoid verbosity in BDD features. This is a
+   * temporary workaround, as in the long term, tenantId would be part of the authenticated user
+   * object and would not have to be supplied as a request parameter
+   */
+  @Value(TENANT_ID_SYSTEM_PROPERTY_EL)
+  private String tenantId;
+
+  /**
+   * @param requestSpecification Request Specification
+   * @param persona Persona
+   * @return RequestSpecification
+   */
+  private static RequestSpecification addPersonaAuthToRequestSpecification(
+      @NonNull RequestSpecification requestSpecification, @NonNull Persona persona) {
```

Comment:
I think the idea was that this may change depending on how auth gets implemented and if it is extracted first the type will only guarantee a string. The intention had been that when we come to implement auth we would only need to update this one method i.e. what happens if `persona.getAuthToken()` no longer returns a string.